### PR TITLE
Tentative deroff test

### PR DESCRIPTION
**Pull request by charliethomson**
There are still a lot of failures, need to figure out why. Currently the test outputs the `get_output()` result of the deroffer to `./test_deroff_output/<manpage_name>.deroffed`, but only if the deroffed was different from the deroffed from the python. It also clones the `manpages` repo (so we have a consistent test suite). There are lots of bugs, somebody please help me find them :sob:
It looks like the majority of the failures are due to the `.de` macro, or `.nr`, but please investigate, i've run out of brain to diag 

🦀

### CODE REVIEW

File: src/deroff.rs [hunk 1]
```diff
@@ -1477,12 +1484,212 @@ fn test_request_or_macro() {
...
+    fs::create_dir(TEST_OUTPUT_DIR)?;
+
+    let stdout = io::stdout();
+    let mut lock = stdout.lock();
+
+    // If we don't have the manpages, get them
+    if !Path::new("./manpages").exists() {
+        print!("Cloning remote manpages... ");
+        lock.flush()?;
+        match Repository::clone("https://github.com/rust-dc/manpages.git", "./manpages") {
```
Comment (pickfire) [inline src/deroff.rs:1515]: I don't think we should be doing this in our test. Maybe we could clone it first as a submodule.

**Comment by pickfire** [comment]
@charliethomson I haven't check this yet but how do you want me to help?

**Comment by pickfire** [comment]
I just realized I have written quite a bit of bugs here and there. Some fixed in https://github.com/rust-dc/fish-manpage-completions/pull/106 but still cannot output to standard output yet.

**Comment by scooter-dangle** [comment]
@charliethomson, do we still want to merge this? I get conflicts and then build errors when I try to resolve the conflicts locally.

**Pull request closed without merging**